src/extensions.mjs

```diff
@@ -0,0 +1,95 @@
+import {
+  WispBuffer,
+  extension_types,
+ encode_text 
+} from "./packet.mjs"
+
+
+export class Extension {
+  constructor({
+    ext_id,
+    buffer
+  }) {
+    this.ext_id = ext_id;
+    this.buffer = buffer;
+  }
+}
+
+export class ExtensionWrapper {
```

Comment:
Why is this a separate class? Can't all this logic be part of the `Extension` base class?